.github/workflows/website-production.yml

```diff
@@ -0,0 +1,95 @@
+name: website-production
+
+defaults:
+  run:
+    working-directory: website
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploy:
+        description: 'type "deploy" to confirm deploy to production (master branch only)'
+        required: false
+  push:
+    branches: [main]
+    paths:
+      - .github/workflows/website.yml
+      - website/**
+      - packages/snack-sdk/**
+      - .eslint*
+      - .prettier*
+      - yarn.lock
+  pull_request:
+    branches: [main]
+    paths:
+      - .github/workflows/website.yml
+      - website/**
+      - packages/snack-sdk/**
+      - .eslint*
+      - .prettier*
+      - yarn.lock
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Decrypt secrets
+        uses: sliteteam/github-action-git-crypt-unlock@a09ea5079c1b0e1887d4c8d7a4b20f00b5c2d06b
+        env:
+          GIT_CRYPT_KEY: ${{ secrets.GIT_CRYPT_KEY }}
+
+      - name: Install Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@94337306dda8180d967a56932ceb4ddcf01edae7
+        with:
+          project_id: exponentjs
+          service_account_key: ${{ secrets.GCLOUD_KEY }}
+          export_default_credentials: true
+
+      - name: Install Google Cloud components
+        run: gcloud components install kubectl skaffold
+
+      - name: Configure Google Cloud SDK
+        run: |
+          gcloud config set project exponentjs
+          gcloud config set compute/zone us-central1-f
+          gcloud config set container/cluster exp-central
+          gcloud container clusters get-credentials exp-central
+          gcloud auth configure-docker gcr.io
+
+      - name: Build and test
+        run: ENVIRONMENT=production skaffold build --filename website/skaffold.yaml --file-output website/build.json
+        working-directory: ./
+
+      - name: Add change cause
+        run: kustomize edit add annotation kubernetes.io/change-cause:"Github Actions deploying $GITHUB_SHA at $(date)"
+        working-directory: website/deploy/production
+
+      - name: Deploy to production
+        if: success() && github.ref == 'refs/heads/main' && github.event.inputs.deploy == 'deploy'
+        run: ENVIRONMENT=production skaffold deploy --filename website/skaffold.yaml --status-check --build-artifacts website/build.json
+        working-directory: ./
+
+      - name: Notify build on Slack
+        uses: 8398a7/action-slack@v3
+        if: always() && github.ref == 'refs/heads/main' && github.event_name != 'pull_request' && github.event.inputs.deploy != 'deploy'
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_SNACK }}
+        with:
+          channel: '#snack'
+          status: ${{ job.status }}
+          author_name: Build Snack website for Production
+          fields: message,commit,author,took
+
+      - name: Notify deploy on Slack
```

Comment:
Isn't this step causing double messages in Slack? (I think it's a duplicate of the previous step)

Reply:
The first one will trigger upon build only. The second one will trigger after approval to be deployed to the production server. See the end of the `if` statements.

```
- name: Notify build on Slack
   if: always() && github.ref == 'refs/heads/main' && github.event_name != 'pull_request' && github.event.inputs.deploy != 'deploy'

- name: Notify deploy on Slack
   if: always() && github.ref == 'refs/heads/main' && github.event_name != 'pull_request' && github.event.inputs.deploy == 'deploy'
```